### PR TITLE
Apply SpecialReport or fallback to the standard behaviour

### PR DESCRIPTION
**Pull request by DiegoVazquezNanini**
## What does this change?
Adds the ability to use the SpecialReport tone using the targeting tool instead of hashing the tag inside facia-scala-client. This greatly simplifies the procedure to apply the tone.
This is the targeting tool: https://targeting.gutools.co.uk/campaigns

## What is the value of this and can you measure success?
We will have less hassle when applying the special report tone. 

## Does this affect other platforms - Amp, Apps, etc?
Nop
## Tested in CODE?
Will test it in CODE

@jfsoul @TBonnin @stephanfowler

### CODE REVIEW

File: common/app/model/content.scala [hunk 1]
```diff
@@ -304,6 +306,17 @@ final case class Content(
   val nonCompliantOutbrainAmp = (hasStoryPackage && tags.series.nonEmpty) || (tags.series.length > 1)
 }
 
+object CardStylePicker {
```
Comment (TBonnin) [inline common/app/model/content.scala]: Is there a better place for this object to live at. This `Content` file is already huge and I am not sure this belongs in it?

File: common/app/model/content.scala [hunk 2]
```diff
@@ -304,6 +306,17 @@ final case class Content(
...
+  def apply(content: contentapi.Content, trailMetaData: MetaDataCommonFields): com.gu.facia.api.utils.CardStyle = {
+    specialReportFromTargeting(content) getOrElse fapiutils.CardStyle(content, TrailMetaData.empty)
+  }
+
+  private def specialReportFromTargeting(content: contentapi.Content): Option[com.gu.facia.api.utils.CardStyle] = {
+    val campaigns = _root_.commercial.targeting.CampaignAgent.getCampaignsForTags(content.tags.map(_.id))
+    campaigns.filterNot(_.fields == ReportFields).map(_ => SpecialReport).headOption
```
Comment (jfsoul) [inline common/app/model/content.scala]: I think I would find a chain here easier to read:

```
private def specialReportFromTargeting(content: contentapi.Content): Option[CardStyle] = 
  CampaignAgent.getCampaignsForTags(content.tags.map(_.id))
    .filterNot(_.fields == ReportFields)
    .map(_ => SpecialReport)
    .headOption
```

what do you think?

File: common/app/model/content.scala [hunk 3]
```diff
@@ -304,6 +306,17 @@ final case class Content(
   val nonCompliantOutbrainAmp = (hasStoryPackage && tags.series.nonEmpty) || (tags.series.length > 1)
 }
 
+object CardStylePicker {
+  def apply(content: contentapi.Content, trailMetaData: MetaDataCommonFields): com.gu.facia.api.utils.CardStyle = {
```
Comment (jfsoul) [inline common/app/model/content.scala]: and if there is a naming clash here or elsewhere (e.g. SpecialReport) just use an alias in the import?

File: common/app/model/CardStylePicker.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package model
+
+import com.gu.facia.api.utils.{CardStyle, SpecialReport}
+import com.gu.facia.client.models.{MetaDataCommonFields, TrailMetaData}
+import com.gu.targeting.client.ReportFields
+import com.gu.contentapi.client.model.{v1 => contentapi}
```
Comment (TBonnin) [inline common/app/model/CardStylePicker.scala]: why not importing `import com.gu.contentapi.client.model.v1.Content` directly

File: common/app/model/CardStylePicker.scala [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+object CardStylePicker {
+
+  def apply(content: contentapi.Content, trailMetaData: MetaDataCommonFields): CardStyle = {
+    specialReportFromTargeting(content) getOrElse CardStyle(content, TrailMetaData.empty)
+  }
+
+  private def specialReportFromTargeting(content: contentapi.Content): Option[CardStyle] = {
+    CampaignAgent.getCampaignsForTags(content.tags.map(_.id))
+      .filterNot(_.fields == ReportFields)
+      .map(_ => SpecialReport)
```
Comment (TBonnin) [inline common/app/model/CardStylePicker.scala]: not sure I follow what's going on here. We are mapping but discarding the initial value?

File: common/app/model/CardStylePicker.scala (hunk 4 above)
Comment (DiegoVazquezNanini) [inline common/app/model/CardStylePicker.scala]: I originally tried that but there is another Content in scope.

File: common/app/model/CardStylePicker.scala (hunk 5 above)
Comment (DiegoVazquezNanini) [inline common/app/model/CardStylePicker.scala]: I think that is right. I will explain IRL.

**Review by TBonnin — APPROVED** [review]

**Pull request merged**